platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,346 @@
+
+Transferring the ability to generate ledger signatures from one set of consensus nodes to another is done by having
+each node generate a `TssMessage` for each share that fractures the share into a number of subshares or `shares of
+shares`. Each `Share of a share` is encrypted with the public key of the node in the next consensus roster that it
+belongs to so that only the intended node can use the `share of a share` to recover that node's share's private key.  
+The collection of `TssMessages` generated from nodes in the previous consensus roster forms the `key material` for
+the new consensus roster. To bootstrap a network and generate the ledger private and public keys, each node creates
+a random share for themselves and generates a `TssMessage` containing `shares of shares` for the total number of shares
+needed in the consensus roster. The use of random shares at the outset creates a random ledger private key that nobody
+knows.
+
+This process of re-keying the next consensus roster during an address book change takes an asynchronous amount of
+time through multiple rounds of consensus to complete. This requires saving incremental progress in the state to ensure
+that the process can resume from any point if a node or the network restarts. Switching to the next consensus roster
+cannot happen until that roster has enough nodes able to recover a threshold number of shares so that an aggregation of
+node signatures can generate the ledger signature.
+
+##### Elliptic Curve Decisions
+
+Our first implementation of Groth21 will use the ALT_BN128 elliptic curve which is in use and verifiable by EVMs.  
+If and when the Ethereum ecosystem adopts BLS12_381, we will likely switch over to that more secure curve.
+
+##### New Elliptic Curve Node Keys
+
+Each node will need a new long-term EC key pair in addition to the existing RSA key pair. The EC key pair will be
+used in the Groth21 algorithm to generate and decrypt `TSS-Messages`. These new EC Keys will not be used for
+signing messages, only for generating the shares. It is the share keys that are used to sign messages. The public
+keys of these long-term node specific EC keys must be in the address book.
+
+##### Groth21 Drawbacks
+
+The Groth21 algorithm is not able to model arbitrary precision proportions of weight assigned to nodes in the
+network. For example if each node in a 4 node network received 1 share, then every share has a weight of 1/4.
+If there are a total of N shares, then the distribution of weight can only be modeled in increments of (1/N) * total
+weight. The cost of re-keying the network in an address book change is quadratic in the number of shares. This forces us
+to pick a number of total shares with a max value in the thousands. This modeling of weight is a discrete using small
+integer precision.
+
+#### Alternatives Considered
+
+The list of options considered here were based off of prototypes developed by Rohit Sinha, the Swirlds Labs
```

Comment:
Does it makes sense to name Rohit Sinha here? if yes, can we link to the GitHub profile?

Reply:
linked to Rohit's profile